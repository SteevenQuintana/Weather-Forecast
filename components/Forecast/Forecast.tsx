import { ForecastProps } from '../../interfaces/cities.interface'
import Highlights from './Highlights/Highlights'
import DayInformation from './DayInformation/DayInformation'

const Forecast = ({ forecast }: ForecastProps) => (
  <div
    className='flex flex-col md:grid sm:grid-cols-[30%_70%] gap-4 mx-2 md:pr-4'
    data-testid='forecast'
  >
    <article>
      <DayInformation forecast={forecast} />
    </article>
    <article>
      <Highlights forecast={forecast} />
    </article>
  </div>
)

export default Forecast
