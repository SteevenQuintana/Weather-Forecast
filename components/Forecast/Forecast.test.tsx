import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import Forecast from './Forecast'
import { Forecast as ForecastInterface } from '@/interfaces/cities.interface'

const mockForecast: ForecastInterface = {
  name: 'City Name',
  country: 'Country Name',
  weather: {
    description: 'Sunny',
    main: 'Clear',
    icon: '01d'
  },
  sunrise: 92291,
  sunset: 91291,
  clouds: 91,
  main: {
    temp: 25,
    temp_max: 30,
    temp_min: 20,
    pressure: 1010,
    humidity: 60,
    feels_like: 26
  },
  wind: {
    speed: 5,
    deg: 90
  }
}

describe('Forecast', () => {
  it('should render successfully', () => {
    render(<Forecast forecast={mockForecast} />)
    expect(screen.getByText(mockForecast.country)).toBeInTheDocument()
    expect(
      screen.getByText(mockForecast.weather.description)
    ).toBeInTheDocument()
  })

  it('should snapshots the Forecast component', () => {
    const { container } = render(<Forecast forecast={mockForecast} />)
    expect(container).toMatchSnapshot()
  })
})
