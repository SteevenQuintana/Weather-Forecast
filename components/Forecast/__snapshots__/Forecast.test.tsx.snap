// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast should snapshots the Forecast component 1`] = `
<div>
  <div
    class="flex flex-col md:grid sm:grid-cols-[30%_70%] gap-4 mx-2 md:pr-4"
    data-testid="forecast"
  >
    <article>
      <div
        class="h-full flex flex-col justify-between items-center text-center bg-[#3054b7ca] p-4 rounded-2xl dark:bg-[#3a4957cc]"
      >
        <h2
          class="text-2md font-bold"
        >
          Today's information
        </h2>
        <div
          class="h-[140px] flex w-[200px] overflow-hidden object-cover"
        >
          <img
            alt="Sunny"
            class="mx-auto my-0 scale-[1.7]"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2F01d%402x.png&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2F01d%402x.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2F01d%402x.png&w=384&q=75 2x"
            style="color: transparent;"
            width="150"
          />
        </div>
        <div
          class="max-h-full h-[340px] w-full"
        >
          <div
            class="h-full flex flex-col justify-between"
          >
            <p
              class="capitalize italic"
            >
              Sunny
            </p>
            <div
              class="flex flex-col"
            >
              <h3
                class="text-[4rem]"
              >
                25
                <sup
                  class="text-[2rem] font-normal"
                >
                  °C
                </sup>
              </h3>
              <p
                class="flex justify-center gap-4"
              >
                <span>
                  H: 
                  30
                   
                  <span>
                    °C
                  </span>
                </span>
                <span>
                  L: 
                  20
                   
                  <span>
                    °C
                  </span>
                </span>
              </p>
            </div>
            <p>
              City Name
              , 
              <span>
                Country Name
              </span>
            </p>
            <p>
              Feels like 
              26
               
              <span>
                °C
              </span>
            </p>
            <div
              class="flex justify-center"
            >
              <button
                class="bg-[#ffffff] dark:bg-[#1c2f3d] text-[#1d1d1d] dark:text-[#dbdbdb] rounded-md py-2 px-7 rounded-[100px]"
                type="button"
              >
                Change to °F
              </button>
            </div>
          </div>
        </div>
      </div>
    </article>
    <article>
      <div
        class="h-full mb-8 text-center flex flex-col bg-[#3054b7ca] p-4 rounded-2xl dark:bg-[#3a4957cc]"
      >
        <h2
          class="text-2md font-bold mb-4"
        >
          Today's Highlights
        </h2>
        <div
          class="grid grid-cols-[repeat(auto-fill,minmax(160px,1fr))] gap-3 h-full"
        >
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Humidity
            </h3>
            <img
              alt="60"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              60
               
              <span>
                %
              </span>
            </p>
          </div>
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Wind Speed
            </h3>
            <img
              alt="5"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              5
               
              <span>
                m/s
              </span>
            </p>
          </div>
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Clouds
            </h3>
            <img
              alt="91"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              91
               
              <span>
                %
              </span>
            </p>
          </div>
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Sunrise
            </h3>
            <img
              alt="20:38"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              20:38
               
              <span>
                a.m.
              </span>
            </p>
          </div>
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Sunset
            </h3>
            <img
              alt="20:21"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              20:21
               
              <span>
                p.m.
              </span>
            </p>
          </div>
          <div
            class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
          >
            <h3>
              Pressure
            </h3>
            <img
              alt="1010"
              class="m-auto"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
            <p>
              1010
               
              <span>
                hPa
              </span>
            </p>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;
